src/RageSoundReader.cpp

```diff
@@ -6,34 +6,34 @@
 REGISTER_CLASS_TRAITS( RageSoundReader, pCopy->Copy() );
 
 /* Read(), handling the STREAM_LOOPED and empty return cases. */
-int RageSoundReader::RetriedRead( float *pBuffer, int iFrames, int *iSourceFrame, float *fRate )
-{
-	/* pReader may return 0, which means "try again immediately".  As a failsafe,
-	 * only try this a finite number of times.  Use a high number, because in
-	 * principle each filter in the stack may cause this. */
-	int iTries = 10;
-	while( --iTries )
-	{
-		if( fRate )
+int RageSoundReader::RetriedRead(float* pBuffer, int iFrames, int* iSourceFrame, float* fRate) {
+	if (iFrames == 0) {
+		return 0;
+	}
+
+	// pReader may return 0, which means "try again immediately". As a failsafe,
+	// only try this a finite number of times. Use a high number, because in
+	// principle each filter in the stack may cause this.
+	std::uint8_t retryCount = 50;
+	while (--retryCount) {
+		if (fRate) {
 			*fRate = this->GetStreamToSourceRatio();
-		if( iSourceFrame )
+		}
+		if (iSourceFrame) {
 			*iSourceFrame = this->GetNextSourceFrame();
-
-		int iGotFrames = this->Read( pBuffer, iFrames );
-
-		if( iGotFrames == RageSoundReader::STREAM_LOOPED )
-			iGotFrames = 0;
-
-		if( iGotFrames != 0 )
-			return iGotFrames;
-
-		// If the user is having I/O issues, give them a hint in the logs.
-		LOG->Warn( "Read() failed, retrying..." );
+		}
+		int framesRead = this->Read(pBuffer, iFrames);
+		if (framesRead == RageSoundReader::STREAM_LOOPED) {
+			framesRead = 0;
+		}
+		if (framesRead != 0) {
+			return framesRead;
+		}
 	}
 
-	LOG->Warn( "Read() returned a failure status after 10 attempts to read the file; likely an I/O error" );
+	LOG->Warn("Fatal error: could not read from the sound file.");
```

Comment:
This isn't a Fatal error since we're doing a Log Warn

Reply:
Good point, sorry about that.